rpcs3/Emu/Cell/Modules/cellPad.cpp

```diff
@@ -235,43 +235,78 @@ s32 cellPadGetData(u32 port_no, vm::ptr<CellPadData> data)
 		btnChanged = true;
 	}
 
-	//not sure if this should officially change with capabilities/portsettings :(
-	data->len = 24;
+	
+	// the real hardware only fills the buffer up to "len" elements (16 bit each)
+	if (pad.m_port_setting & CELL_PAD_SETTING_SENSOR_ON)
+	{
+		// report back new data every ~10 ms even if the input doesn't change
+		// this is observed behaviour when using a Dualshock 3 controller
+		static std::chrono::time_point<steady_clock> last_update = steady_clock::now();
```

Comment:
This shouldnt be here and needs to be specific to the pad/port number, the game doing something like this
```
cellPadGetData(1, data);
cellPadGetData(2, data);
```
will return no data for the second port with your code

Reply:
Yes, I mentioned this problem above. The actual input providers (or an abstraction layer in between) could do the job better. Would also probably emulate the real system better since its likely that the controller decides when the data changed (to save energy) instead of cellPadGetData.